src/client/initialize.ts

```diff
@@ -135,6 +136,18 @@ export function initialize<T = {}>(): PluginInstance<T> {
       ) {
         void execPromise('wb:plugin:selection:set', id, elementId, selection);
       },
+      registerActionTrigger(id: string) {
```

Comment:
This would work, but it's a little different then the pattern we have for other APIs. Could we instead for registering/unregistering rely on a listener like those on lines 50-58? Otherwise we wouldn't need the change to `CustomPluginConfigOptions` in `types.ts`

Reply:
I don’t think event listeners are needed here on the plugin side, because for `action-trigger`, plugin only needs to send a message to Slate every time the action gets triggered, and Slate will then handle the event (trigger all matching action definitions). Plugin doesn't need to hear back or receive any updates from Slate (please correct me if I'm wrong). Whereas for `action-effect`, plugin should have some event listener and handler, but I’m not working on it yet.

Reply:
I don’t think event listeners are needed here on the plugin side, because for `action-trigger`, plugin only needs to send a message to Slate every time the action gets triggered, and Slate will then handle the event (trigger all matching action definitions). Plugin doesn't need to hear back or receive any updates from Slate (please correct me if I'm wrong). Whereas for `action-effect`, plugin should have some event listener and handler, but I’m not working on it yet.

Reply:
We want this `actionsTriggerIds` object to only include registered triggers (i.e. those that exist in the app and can be seen by the user), and the core functionality of this registration shouldn't take place within the plugin. If we'd want to display all of the registered triggers in the UI, we'd need some way for the plugin to agree with the app on what triggers are registered and are functional. The current mechanism for this with other APIs is the config, which is essentially the plugin "requesting" certain triggers (or some other entity) be registered. The triggers should be registered within the app, then there should be some confirmation sent back to the plugin with the registered ids.

Reply:
I see, totally makes sense. Just to be clear, is the entire process 
1. plugin requests triggers be registered in slate by sending a message (eg. `wb:plugin:trigger:register` with the triggerIds to be registered) 
2. slate registers the triggers and sends back a confirmation (like `wb:plugin:trigger:update`) 
3. plugin stores the newly registered triggers into `actionTriggerIds` which can be used to check if the trigger is already registered in slate, and if it is, plugin will send slate a message (like `wb:plugin:trigger:invoke`) to trigger the action?

Reply:
I don't think slate needs to tell the plugin that an action trigger has been registered. I think registering triggers should be similar to how the plugin configuration options are initialized. Like, they are just there once you load the plugin.

Reply:
I personally like the idea of the plugin having knowledge that their config object has been registered, since if we want any expansions, where the plugin wants information about its subscribed actions, then there's a good pattern for fetching it -- but that step isn't 100% necessary for this project.

If we don't go with that step, then we have to be careful to create all the ids as needed when the plugin initializes and keep that up-to-date within the plugin as the plugin's config changes, which we don't have a pattern for at the moment. I'm just concerned that this will wind up increasing the work required.

Reply:
I think you're definitely over complicating how plugin action triggers get configured and initialized. A plugin action trigger ID should be the same as any other config item. If the user wants to respond to config updates, then they can do so using the existing `config.subscribe` method.

Reply:
The issue here is that we're not storing the action directly in the config object (though we can change other APIs if that'd be an easier experience - I think rethinking the general API for plugins to make the experience easier could be helpful), so we can't just call `config.subscribe`. I'm happy to go with either approach for this project, whichever seems simpler since they both are completely feasible.